internal/contour/handler.go

```diff
@@ -267,3 +269,28 @@ func (e *EventHandler) setStatus(statuses map[dag.Meta]dag.Status) {
 		}
 	}
 }
+
+// WaitForCacheProcess waits for the Cache to be processed by checking the
+// value of "outstanding"
+// Return true if processed fully or false if timed out waiting
+func (e *EventHandler) WaitForCacheProcess(timeout time.Duration) bool {
+	ch := make(chan bool, 1)
+
+	go func() {
+		for e.outstanding > 0 {
+			time.Sleep(100 * time.Millisecond) // ZzzzZZzzzz
+		}
+		ch <- true
```

Comment:
you don't need to send on this channel, closing it is enough to make it readable by line 291